packages/atomic/src/components/commerce/facets/atomic-commerce-timeframe-facet/atomic-commerce-timeframe-facet.tsx

```diff
@@ -0,0 +1,317 @@
+    return !this.hasInputRange && !!this.valuesToRender.length;
+  }
+
+  private get shouldRenderFacet() {
+    return this.shouldRenderInput || this.shouldRenderValues;
+  }
+
+  private get shouldRenderInput() {
+    const {
+      firstSearchExecuted,
+      hasError,
+      isLoading,
+      hasProducts: hasResults,
+    } = this.summary.state;
+    return shouldDisplayInputForFacetRange({
+      hasInputRange: this.hasInputRange,
+      searchStatusState: {
+        firstSearchExecuted,
+        hasError,
+        hasResults,
+        isLoading,
+      },
+      facetValues: this.facet?.state?.values || [],
+      hasInput: true,
+    });
+  }
+
+  private get hasValues() {
+    if (this.facet?.state.values.length) {
+      return true;
+    }
+
+    return !!this.valuesToRender.length;
+  }
+
+  private get numberOfSelectedValues() {
+    if (this.hasInputRange) {
+      return 1;
+    }
+
+    return (
+      this.facet?.state.values.filter(({state}) => state === 'selected')
+        .length || 0
+    );
+  }
+
+  private get hasInputRange() {
+    return !!this.inputRange;
+  }
+
+  public disconnectedCallback() {
+    if (this.host.isConnected) {
+      return;
+    }
+  }
+
+  private get isHidden() {
+    return !this.shouldRenderFacet;
+  }
+
+  private resetRange() {
+    this.inputRange = undefined;
+    this.facet.setRanges([]);
+  }
+
+  private registerFacetToStore() {
+    const facetInfo: FacetInfo = {
+      label: () => this.bindings.i18n.t(this.displayName),
+      facetId: this.facet.state?.facetId,
+      element: this.host,
+      isHidden: () => this.isHidden,
+    };
+
+    this.bindings.store.registerFacet('dateFacets', {
+      ...facetInfo,
+      format: (value) => this.formatFacetValue(value),
+    });
+
+    initializePopover(this.host, {
+      ...facetInfo,
+      hasValues: () => this.hasValues,
+      numberOfActiveValues: () => this.numberOfSelectedValues,
+    });
+  }
+
+  private formatFacetValue(facetValue: DateFacetValue) {
+    try {
+      const startDate = deserializeRelativeDate(facetValue.start);
+      const relativeDate =
+        startDate.period === 'past'
+          ? startDate
+          : deserializeRelativeDate(facetValue.end);
+
+      return this.bindings.i18n.t(
+        `${relativeDate.period}-${relativeDate.unit}`,
+        {
+          count: relativeDate.amount,
+        }
+      );
+    } catch (error) {
+      return this.bindings.i18n.t('to', {
+        start: parseDate(facetValue.start).format('YYYY-MM-DD'),
+        end: parseDate(facetValue.end).format('YYYY-MM-DD'),
+      });
+    }
+  }
+  private renderValues() {
+    return this.renderValuesContainer(
+      this.valuesToRender.map((value) => this.renderValue(value))
+    );
+  }
+
+  private renderValue(facetValue: DateFacetValue) {
+    const displayValue = this.formatFacetValue(facetValue);
+    const isSelected = facetValue.state === 'selected';
+    const isExcluded = facetValue.state === 'excluded';
```

Comment:
This makes me think: is there any way for the user to exclude a facet value in the UI?